server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -212,12 +216,14 @@
                         if (failClosed) {
                             throw new IndexClosedException(index.getIndex());
                         } else {
-                            if (options.forbidClosedIndices() == false) {
+                            if (options.forbidClosedIndices() == false && addIndex(index, context, originalIndexExpression)) {
```

Comment:
question: given that only search allows to provide the new flag, but search only works against open indices by definition, is it necessary to check the metadata and honor the option for closed indices? I guess this second condition is always true? Or do you have in mind cases where ignore_throttled may be used against closed indices as well?

Reply:
it's not a big deal, only a bit confusing to me, because on one hand we don't expose this option to all the API, on the other hand we do apply it to all the API and even to closed indices. I think it's good as-is, let's maybe add a comment to explain this?

Reply:
maybe let's make sure we have a test that exercises the second if in a scenario where it returns false?

Reply:
++